beacon_node/network/src/beacon_processor/worker/gossip_methods.rs

```diff
@@ -1809,30 +1809,13 @@ impl<T: BeaconChainTypes> Worker<T> {
                 );
 
                 // Compute the slot when we received the message.
-                let received_slot = self
-                    .chain
-                    .slot_clock
+                let seen_clock = &self.chain.slot_clock.freeze_at(seen_timestamp);
+                let received_slot = seen_clock
                     .slot_of(seen_timestamp)
                     .unwrap_or_else(|| self.chain.slot_clock.genesis_slot());
 
-                // The message is "excessively" late if it was more than one slot late.
-                let excessively_late = received_slot > sync_committee_message_slot + 1;
-
-                // This closure will lazily produce a slot clock frozen at the time we received the
-                // message from the network and return a bool indicating if the message was invalid
-                // at the time of receipt too.
-                let invalid_in_hindsight = || {
-                    let seen_clock = &self.chain.slot_clock.freeze_at(seen_timestamp);
-                    let hindsight_verification =
-                        sync_committee_verification::verify_propagation_slot_range(
-                            seen_clock,
-                            &sync_committee_message_slot,
-                        );
-                    hindsight_verification.is_err()
-                };
-
                 // Penalize the peer if the message was more than one slot late
-                if excessively_late && invalid_in_hindsight() {
+                if received_slot > sync_committee_message_slot + 1 {
```

Comment:
I don't understand this

The previous definition of `excessively_late` was `received_slot > sync_committee_message_slot + 1`, and we dropped the `&& invalid_in_hindsight()`. The previous definition of `received_slot` used the regular slot clock, but I think it would compute the same slot as the frozen slot clock, i.e. `seen_clock.slot_of(seen_timestamp) == slot_clock.slot_of(seen_timestamp)`. So the first part of the condition is equivalent?

Dropping the `invalid_in_hindsight` check then seems to _widen_ the scope for this penalty to apply. We will start penalising messages that arrived in the first 500ms of `sync_committee_message_slot + 2`, which would _not_ be `invalid_in_hindsight()` but _will_ be `excessively_late` because `received_slot` will equal `sync_committee_message_slot + 2`.

What am I missing? Can you describe the case where we were behaving incorrectly in more detail please? :pray:

Reply:
Oh you're right, sorry I made a mistake. I'll close this.

Reply:
All good, I went back and forth trying to convince myself it was equiv before I realised the first 500ms case

Reply:
Past-us did a good job of writing and reviewing :sweat_smile:

Reply:
Wait a sec, you were right! Sync committee messages are only on time in the _current_ slot, so my thing about a message in the first 500ms of `sync_committee_message_slot + 2` doesn't apply. The spec is:

> [IGNORE] The message's slot is for the current slot (with a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance), i.e. sync_committee_message.slot == current_slot.